### ISSUE TITLE
IStaticDataSource with AES 256 Encryption

### ISSUE THREAD

**Issue by skhilliard** [icsharpcode/SharpZipLib#385]
### Steps to reproduce
1.  Define a simple data source that implements the IStaticDataSource interface:

```cs
public class CustomStaticDataSource : IStaticDataSource
{
	private Stream _stream;

	// Implement method from IStaticDataSource
	public Stream GetSource()
	{
		return _stream;
	}

	// Call this to provide the memorystream
	public void SetStream(Stream inputStream)
	{
		_stream = inputStream;
		_stream.Position = 0;
	}
}
```

2. Use this data source to add a new ZIP entry to a ZipFile whose password is set using this method:

`void Add(IStaticDataSource dataSource, ZipEntry entry);`

Where the entry has the desired name and entry.AESKeySize is set to 256.

### Expected behavior
Zip file is produced that is password protected using AES encryption.

### Actual behavior
Zip file is produced, but using weaker ZipCrypto encryption.

### Version of SharpZipLib
1.2.0 (for .NET Core)

### Obtained from (only keep the relevant lines)
- Package installed using NuGet

**Comment by Numpsy** [comment]
ZipFile is missing some pieces it needs to support AES encryption of entries (was it actually possible to request it to prior to 1.2.0?).

I started looking at filling in the holes, but haven't had time to get it done.

**Comment by Numpsy** [comment]
@piksel As long as ZipFile doesn't support the creation of entries with AES encryption, would it be better if the (IStaticDataSource dataSource, ZipEntry entry) version of Add threw when you tried to add such an entry instead of silently not using AES?

**Comment by piksel** [comment]
Yeah, I agree.

**Comment by piksel** [comment]
I am closing this as it should have been fixed by #420. Note that ZipFile still doesn't support AES, but at least now it won't silently use another encryption.
